File: java/spring_quiz/src/main/java/com/securefromscratch/quiz/safety/FullName.java
package com.securefromscratch.quiz.safety;

import io.swagger.v3.oas.annotations.media.Schema;

import static com.securefromscratch.quiz.utils.StringFuncs.*;

public class FullName {
    public static final String VALID_CHARS = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ'.()";
    public static final int MIN_LENGTH = 3;
    public static final int MAX_LENGTH = 40;

    private String m_value;

    public FullName(String value) {
        int spaceIdx = value.indexOf(' ');
        if (spaceIdx == -1) {
            throw new IllegalArgumentException(value);
        }

        m_value = value;
    }

    public String get() {
        return m_value;
    }

    @Override
    public String toString() {
        return get();
    }
}

File: csharp/dotnetcore_quiz/Safety/FullName.cs
namespace SecureFromScratch.Quiz.Safety
{
    public class FullName
    {
        
        public string Username { get; }

        public FullName(string username)
        {
            if (string.IsNullOrWhiteSpace(username))
                throw new ArgumentException("Username cannot be empty.");
            Username = username;
        }

        public override string ToString() => Username;
    }
}
